src/config/mnemonic.ts

```diff
@@ -49,6 +51,10 @@ export const configExtender: ConfigExtender = (finalConfig, userConfig) => {
         mnemonic: defaultMnemonic
       }
     }
+
+    // Since finalNetwork is new reference (due to { ... }) re-assign to finalConfig
+    if (!finalConfig.networks) finalConfig.networks = {}
+    finalConfig.networks[networkName] = finalNetwork
   }
```

Comment:
Just wondering, would be it be worth returning the `finalConfig`? I do find `configExtender()`'s API a bit... unsanitary as it writes to the passed in object as opposed to just making a clone and returning it.

Reply:
We decided to use `configExtener()` with the pattern chosen by the builder team to create the `BuidlerConfig` object. They use the `ConfigExtender` type that has the following types:
```
type ConfigExtender = (config: ResolvedBuidlerConfig, userConfig: DeepReadonlyObject<BuidlerConfig>) => void
```

Reply:
Ah cool, yes, makes sense to stick with their pattern!

If that type isn't coming directly from a buidler import, could we leave a note that we're mimicking their "canonical" style here?

Reply:
Everywhere we are using it we export the `ConfigExtender` type from builder. I think is quite straight forward already.